File: Sorting/Insertionsort.cpp
//
// Created by Tay on 10/11/17.
//

void insertionSort(int table[], int size) {
    int key, j;
    for (int i = 1; i < size; i++) {
        key = table[i];
        j = i - 1;
        while (j >= 0 && table[j] > key) {
            table[j + 1] = table[j];
            j--;
        }
        table[j + 1] = key;
    }
}

int main() {
    int table[] = {13, 32, 43, 14, 25};
    insertionSort(table, 5);
    return 0;
}

File: Sorting/insertionsort.java
/* Created by Jacob Rodgers
   10/11/17
 */

 public class insertionSort {

     public void iSort(int userArray[]){
         for (int i = 1; i < userArray.length; i++){
             //create an index variable for tracking current number
             int index = userArray[i];
             //and a second counting variable, j
             int j = i - 1;

             //Iterate through entire Array comparing each variable to the index assigned above
             while ((j >= 0) && (userArray[j] > index)) {
                 //If there is a number out of order then we will switch them
                 userArray[j+1] = userArray[j];
                 j = j - 1;
             }
             userArray[j+1] = index;
         }
     }
 }





